libqtile/widget/thermal_zone.py

```diff
@@ -0,0 +1,29 @@
+from libqtile.widget import base
+
+class ThermalZone(base.ThreadPoolText):
+    """
+    This is simple thermal zone monitoring widget.
+
+    Widget requirements: linux kernel >= 5.x.x
+
+    It allows you to get CPU temperature in Celsius degrees from /sys/class/thermal/ sensors.
+    """
+    orientations = base.ORIENTATION_HORIZONTAL
+    defaults = [
+        ('update_interval', 2.0, 'Update interval for the thermal zone sensor'),
+        ('zone', '0', 'Default thermal zone'),
+    ]
+
+    def __init__(self, **config):
+        super().__init__("", **config)
+        self.add_defaults(ThermalZone.defaults)
+
+    def get_temp(self):
+        with open('/sys/class/thermal/thermal_zone' + self.zone + '/temp') as f:
+            value = str(round(int(f.read().rstrip()) / 1000)) + '°C'
```

Comment:
There needs to be some error handling, e.g. if the path is invalid or if there's an invalid value.

Reply:
Done

Reply:
Could you perhaps catch an exception and handle the error that way? The rationale of that approach is that exceptions will be rare, so we can get away with skipping the `path.isfile()` check because it will almost always be fine. If we wrap the `with open` block in a `try...catch` then we remove that (admittedly minor) overhead of the check every 2 seconds.

Reply:
Done